### ISSUE TITLE
EditDistance$EditMatrix performance issues

### ISSUE THREAD

**Issue by GoogleCodeExporter** [etorreborre/specs#110]
```
It seems to me like the implementation of EditDistance either needs to be
made to perform better (in terms of memory and execution speed) or it needs
to be disabled by default for Strings of medium size or bigger. I got an
OOM with a heap of 3GB while comparing about a page of XML output[1]. I've
also seen cases where comparisons that took less than a second took a very
long time.

[1] org.specs.runner.UserError: java.lang.OutOfMemoryError: Java heap space
    at org.specs.util.EditDistance$EditMatrix.<init>(EditDistance.scala:48)
    at org.specs.matcher.AnyBaseMatchers$$anon$2.apply(AnyMatchers.scala:97)
    at org.specs.specification.Expectable.executeMatch$1(Expectable.scala:100)
    at org.specs.specification.Expectable.applyMatcher(Expectable.scala:115)
    at org.specs.specification.Expectation.must(Expectable.scala:169)
```

Original issue reported on code.google.com by `ismaelj` on 17 Jan 2010 at 2:01

**Comment by GoogleCodeExporter** [comment]
```
I forgot to mention that the stacktrace above is for specs 1.6.1.
```

Original comment by `ismaelj` on 17 Jan 2010 at 2:02

**Comment by GoogleCodeExporter** [comment]
```
I tried to fix that issue by making comparisons using the edit distance work on 
strings 
splitted on newlines.

I'm also thinking that comparisons on xml files should rather be some sort of 
"tree 
edit distance".

Tell me if that suits you and I will publish the official 1.6.2 version soon.
```

Original comment by `etorrebo...@gmail.com` on 26 Jan 2010 at 3:24
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Sorry for the delay, I've been trying to make time to try this, but it's been 
very
busy. I see that you have released 1.6.2, that's good. I will file a new issue 
if I
am still not happy. :) Thanks!
```

Original comment by `ismaelj` on 27 Jan 2010 at 9:22

**Comment by GoogleCodeExporter** [comment]
Original comment by `etorrebo...@gmail.com` on 14 Feb 2010 at 2:43
- Changed state: **Verified**
